src/main/java/org/opentripplanner/updater/GraphUpdaterManager.java

```diff
@@ -189,18 +189,29 @@ public ScheduledExecutorService getScheduler() {
         return scheduler;
     }
 
-    private void updateInternalState() {
-        try {
-            if(numberOfNonePrimedUpdaters.intValue() > 0) {
-                int waiting = (int) updaterList.stream().filter(u -> !u.isPrimed()).count();
-                numberOfNonePrimedUpdaters.set(waiting);
-                if(waiting == 0) {
-                    LOG.info("All updaters initialized");
+    /**
+     * This method start a task during startup, witch update the 'numberOfNonePrimedUpdaters'. It
+     * does so in its own thread, using busy-wait(anti-pattern). The ideal would be to add a
+     * callback from each updater to notify the manager about 'isPrimed'. But, this is simple, the
+     * thread is mostly idle, and it is short-lived, so the busy-wait is a compromise.
+     */
+    private void reportReadinessForUpdaters() {
+        Executors.newSingleThreadExecutor().submit(() -> {
+            while (true) {
+                try {
+                    int waiting = (int) updaterList.stream().filter(u -> !u.isPrimed()).count();
+                    numberOfNonePrimedUpdaters.set(waiting);
+                    if (waiting == 0) {
+                        LOG.info("All updaters initialized");
+                        return;
+                    }
+                    //noinspection BusyWait
+                    Thread.sleep(1000);
```

Comment:
Is one second appropriate here?

Reply:
I think so - normally, the updaters should be initialized/primed within a few seconds.